lib/vanagon/component/dsl.rb

```diff
@@ -316,8 +316,12 @@ def add_source(url, ref: nil, sum: nil)
       # @param mode [String] octal mode to apply to the directory
       # @param owner [String] owner of the directory
       # @param group [String] group of the directory
-      def directory(dir, mode: nil, owner: nil, group: nil)
-        @component.directories << Vanagon::Common::Pathname.new(dir, mode: mode, owner: owner, group: group)
+      def directory(dir, mode: nil, owner: nil, group: nil, wix_id: nil)
```

Comment:
Yes - it is a bit overly complex - GUID's are sometimes used because they are unique, but you can use any string that is unique within the whole wix project. The string is of [Microsoft Type "Identifier"](https://msdn.microsoft.com/library/aa369212.aspx) therefore must consist of Alphanumeric characters or the "." and "_" characters and must begin with a letter or underscore.